extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -64,23 +62,26 @@ private function getSmartBannerConfig() {
 	 */
 	private function getTopContributorsPerArticle( $articleId ) {
 		$usersIds = $this->mercuryApi->topContributorsPerArticle( $articleId, self::NUMBER_CONTRIBUTORS );
+
 		return $usersIds;
 	}
 
 	/**
 	 * @desc returns article details
 	 *
-	 * @param int $articleId
+	 * @param Article $article
+	 *
 	 * @return mixed
 	 */
-	private function getArticleDetails( $articleId ) {
-		$articleDetails = $this->sendRequest( 'ArticlesApi', 'getDetails', [ 'ids' => $articleId ] )
-			->getData()[ 'items' ][ $articleId ];
+	private function getArticleDetails( Article $article ) {
+		$articleId = $article->getID();
+		$articleDetails =
+			$this->sendRequest( 'ArticlesApi', 'getDetails', [ 'ids' => $articleId ] )->getData()['items'][$articleId];
```

Comment:
it would be nice to have check for items and articleId keys so we won't throw Fatals if ['items'] is null and we are trying to access key on it